misc.cpp

```diff
@@ -26,11 +26,22 @@ void update_temperature(void)
 void get_serial_number(unsigned int devIdx, char* serial)
 {
 #if (ENABLE_NVML==1)
-    nvmlDevice_t devHandle;
-    NVML_CHECK(nvmlDeviceGetHandleByIndex( devIdx, &devHandle ));
+    try
+    {
+        nvmlDevice_t devHandle;
+        NVML_CHECK(nvmlDeviceGetHandleByIndex( devIdx, &devHandle ));
 
-    unsigned int serialLength = NVML_DEVICE_SERIAL_BUFFER_SIZE;
-    NVML_CHECK(nvmlDeviceGetSerial( devHandle, serial, serialLength ));
+        unsigned int serialLength = NVML_DEVICE_SERIAL_BUFFER_SIZE;
```

Comment:
constant is [30](https://docs.nvidia.com/deploy/nvml-api/group__nvmlConstants.html#group__nvmlConstants_1gca7e1ec7e1c14c01b7980b38188245d0)

Save [`std::strncpy`](https://en.cppreference.com/w/cpp/string/byte/strncpy) used below with a string that is 28 chars and is therefore usually not cropped. Null-added manuall in case it gets cropped.